From 59c99d8382320fa37ecde0c2e7bca3a01545d849 Mon Sep 17 00:00:00 2001
From: George Mocanu <george.mocanu@nxp.com>
Date: Thu, 7 Nov 2024 11:59:06 +0200
Subject: [PATCH] arm64: dts: s32g: Add GoldVIP-specific specs

GoldVIP slightly deviates from the default device tree specification
included in the NXP Linux BSP. Create an include file that contains
the required updates:
- Update the reserved memory ranges for the PFE buffers.
- Add the SRAM memory range used for IPC.
- Add the MC_ME node to provide support to their registers from
user-space.
- Disable various nodes taking in consideration the nodes
configured/used by the real-time domain

Signed-off-by: George Mocanu <george.mocanu@nxp.com>
---
 .../dts/freescale/s32g-goldvip-pfe-slave.dtsi | 10 ++++
 .../boot/dts/freescale/s32g-goldvip.dtsi      | 60 +++++++++++++++++++
 .../dts/freescale/s32g274a-rdb2-pfems.dts     |  1 +
 .../boot/dts/freescale/s32g274a-rdb2.dts      |  1 +
 .../dts/freescale/s32g399a-rdb3-pfems.dts     |  1 +
 .../boot/dts/freescale/s32g399a-rdb3.dts      |  1 +
 6 files changed, 74 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/s32g-goldvip-pfe-slave.dtsi
 create mode 100644 arch/arm64/boot/dts/freescale/s32g-goldvip.dtsi

diff --git a/arch/arm64/boot/dts/freescale/s32g-goldvip-pfe-slave.dtsi b/arch/arm64/boot/dts/freescale/s32g-goldvip-pfe-slave.dtsi
new file mode 100644
index 000000000000..af99bc6ae340
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/s32g-goldvip-pfe-slave.dtsi
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/*
+ * Copyright 2024 NXP
+ *
+ */
+#include "s32g-goldvip.dtsi"
+
+&serdes1 {
+	status = "disabled";
+};
diff --git a/arch/arm64/boot/dts/freescale/s32g-goldvip.dtsi b/arch/arm64/boot/dts/freescale/s32g-goldvip.dtsi
new file mode 100644
index 000000000000..4822b1f84348
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/s32g-goldvip.dtsi
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/*
+ * Copyright 2024 NXP
+ *
+ */
+/ {
+	reserved-memory {
+		/delete-node/ pfebufs@34000000;
+		/delete-node/ pfebufs@34080000;
+
+		pfe_reserved_bmu2: pfebufs@34400000 {
+			compatible = "nxp,s32g-pfe-bmu2-pool";
+			/* BMU2: 512 KB */
+			reg = <0 0x34400000 0 0x80000>;
+			no-map;
+			status = "okay";
+		};
+
+		pfe_reserved_rt: pfebufs@34480000 {
+			compatible = "nxp,s32g-pfe-rt-pool";
+			/* RT: 128 KB */
+			reg = <0 0x34480000 0 0x20000>;
+			no-map;
+			status = "okay";
+		};
+
+		ipcf_shm: ipcf_shm@34520000 {
+			compatible = "nxp,s32g-ipcf-shm";
+			/* 896 KB */
+			reg = <0x0 0x34520000 0x0 0xe0000>;
+			no-map;
+			status = "okay";
+		};
+	};
+
+	soc {
+		mc_me: mc_me@40088000 {
+			compatible = "nxp,s32g-mc_me", "syscon";
+			reg = <0x0 0x40088000 0x0 0x3000>;
+		};
+	};
+};
+
+&pfe_aux0 {
+	/delete-property/ nxp,pfeng-hif-channels;
+	nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_2>;
+	status = "okay";
+};
+
+&stm7 {
+	status = "disabled";
+};
+
+&can_ts_ctrl {
+	status = "disabled";
+};
+
+&spi1 {
+	status = "disabled";
+};
diff --git a/arch/arm64/boot/dts/freescale/s32g274a-rdb2-pfems.dts b/arch/arm64/boot/dts/freescale/s32g274a-rdb2-pfems.dts
index c012875f7339..760172ebfcc2 100644
--- a/arch/arm64/boot/dts/freescale/s32g274a-rdb2-pfems.dts
+++ b/arch/arm64/boot/dts/freescale/s32g274a-rdb2-pfems.dts
@@ -9,6 +9,7 @@
 
 #include "s32g274a-rdb2.dtsi"
 #include "s32g-pfe-slave.dtsi"
+#include "s32g-goldvip-pfe-slave.dtsi"
 
 &pfe_aux0 {
 	status = "okay";
diff --git a/arch/arm64/boot/dts/freescale/s32g274a-rdb2.dts b/arch/arm64/boot/dts/freescale/s32g274a-rdb2.dts
index fa579134532a..b9eeff299e91 100644
--- a/arch/arm64/boot/dts/freescale/s32g274a-rdb2.dts
+++ b/arch/arm64/boot/dts/freescale/s32g274a-rdb2.dts
@@ -8,3 +8,4 @@
 
 /dts-v1/;
 #include "s32g274a-rdb2.dtsi"
+#include "s32g-goldvip.dtsi"
diff --git a/arch/arm64/boot/dts/freescale/s32g399a-rdb3-pfems.dts b/arch/arm64/boot/dts/freescale/s32g399a-rdb3-pfems.dts
index dfe036bc5afe..97dc62eb1f1d 100644
--- a/arch/arm64/boot/dts/freescale/s32g399a-rdb3-pfems.dts
+++ b/arch/arm64/boot/dts/freescale/s32g399a-rdb3-pfems.dts
@@ -9,6 +9,7 @@
 
 #include "s32g399a-rdb3.dtsi"
 #include "s32g-pfe-slave.dtsi"
+#include "s32g-goldvip-pfe-slave.dtsi"
 
 &qspi {
 	status = "okay";
diff --git a/arch/arm64/boot/dts/freescale/s32g399a-rdb3.dts b/arch/arm64/boot/dts/freescale/s32g399a-rdb3.dts
index af80bed4a200..88928e758579 100644
--- a/arch/arm64/boot/dts/freescale/s32g399a-rdb3.dts
+++ b/arch/arm64/boot/dts/freescale/s32g399a-rdb3.dts
@@ -6,3 +6,4 @@
  */
 
 #include "s32g399a-rdb3.dtsi"
+#include "s32g-goldvip.dtsi"
-- 
2.25.1

